packages/filebrowser/src/listing.ts

```diff
@@ -514,6 +514,22 @@ export class DirListing extends Widget {
       });
   }
 
+  /**
+   * Select all listing items in the current directory.
+   */
+  selectAll(): void {
```

Comment:
Can we make it an async method? It does not need to be one as of today, but I have in the back of my mind that we were talking about lazy fetching of contents for large directories and in these cases we may need to call API async, it will be easier to make this change if the methods are already `async`.

```suggestion
  async selectAll(): Promise<void> {
```

Also, most of the surrounding methods (`download`, `goUp`, `duplicate`, `delete`, etc) are already `async`.